README.md

```diff
@@ -1,294 +1,303 @@
+
+- [ ] Following the patterns we used above, build the Signup feature.
+
+---
+
+- [ ] When a user logs in or signs up, we should bring them to the `/users` route. Update both features so that the user is moved to that route after a successful login/signup.
+
+---
+
+- [ ] Try logging out and then go directly to the `/users` route.
+
+* **Question:** What happens? What _should_ happen?
+
+* **Answer:**
+
+---I see the `/users` page as if i was authenticated. Because i am not authenticated, i should not be authorized to see this page.
+
+- [ ] Try _replacing_ the `/users` Route in `App.js` with the following:
+  ```jsx
+  <Route path='/users' render={() => {
+    return this.state.currentUserId ? <UsersContainer /> : <Redirect to='/login' />
+  }} />
+  ```
+
+* **Question:** Describe what is happening in the code above.
+
+* **Your Answer:**
+
+---Upon accessing `/users` route, the App component's state is checked for currentUserId. If currentUserID is present,  the app will render the UsersContainer component. If it is not present, the login component is instead rendered as the user is not authenticated, and thus not authorized to view the page.
+
+- [ ] Now try logging in. Then, when you're on the `/users` page, refresh the page.
+
+* **Question:** What happens and why?
+
+* **Your Answer:**
+
+--- I am not sure i fully grasp this quite yet, but i believe it is because the page is refreshed on the `/users` route before the app component is mounted and currentUserID is not yet set into state. because of this, the users page cannot be shown to the user as per the route, we are redirecting to login page when no currentUserID is found. it seems that the `/users` route execution precedes `componentDidMount()`. Is this correct?
+
+- [ ] To solve this problem, let's add a `loading` key to our App's state, with the default value set to `true`. When `componentDidMount()` finishes, set the `loading` key to equal `false`. Using this key, solve the issue of refreshing on the `/users` page. Make sure everyting continues to work whether you are logged in or out.
+
+* **Question:** What did you do to solve this problem?
+
+* **Your Answer:**
+
+---After setting `loading: true` in app component's state, i added a statement in `componentDidMount()` to set `loading: false` in the component's state once mount has completed.
+
+- [ ] We will have the same problem on the `/users/<userId>/posts` page. Use the same strategy to have this page load correctly on refresh.
+
+* **Question:** In what component did you add the `loading` property and why?
+
+* **Your Answer:**
+
+---I am a little confused on this question as we skipped past this step in the in-class exercise, and now after our other changes made i am not able to reproduce a similar experience to that of the above scenario with `/users`. Would this need to be added in a similar fashion via `componentDidMount()` on the posts component?

```

Comment:
A lot of this depends on our app building approach. We could delay rendering everything in the app, which would fix all the routes but provide less feedback to the user. Or still handle the loading in app but pass the variable to child components such as user container and posts and let them decide whether they need to wait for more data before rendering or not.